data/store/appStore.ts

```diff
@@ -85,3 +86,29 @@ export const useAppStore = create<AppStoreType>()(
     }
   )
 );
+
+/**
+ * Utility function to wait for the app store to be hydrated.
+ * This can be used, for example, to ensure that the xmtp clients have been instantiated and the
+ * conversation list is fetched before navigating to a conversation.
+ *
+ * As of December 19, 2024, when we say Hydration, we mean that the following are true:
+ * 1) XMTP client for all accounts added to device have been instantiated and cached
+ * 2) Conversation list for all accounts added to device have been fetched from the network and cached
+ * 3) Inbox ID for all accounts added to device have been fetched from the network and cached
+ *
+ * You can observe that logic in the HydrationStateHandler, and that will likely be moved once
+ * we refactor accounts to be InboxID based in upcoming refactors.
+ */
+export const waitForXmtpClientHydration = async () => {
+  logger.debug(
+    `[waitForAppStoreHydration] Waiting for app store hydration to complete`
+  );
+  while (!useAppStore.getState().hydrationDone) {
+    logger.debug(
+      `[waitForAppStoreHydration] App store hydration not complete, waiting 100ms...`
+    );
+    await new Promise((resolve) => setTimeout(resolve, 100));
```

Comment:
Why are you doing this vs creating a subscription to the store instead?

Reply:
Yes I thought about that too. But It would look something like that no? Which isn't cleaner IMO or even much more performant. Maybe we keep what Michael have for now which is good enough IMO.

And we'll probably have to refactor stuff anyway later


```
const waitForReady = (): Promise<void> => {
  return new Promise((resolve) => {
    const { isReady } = useAccountsStore.getState();
    if (isReady) {
      resolve();
      return;
    }
    const unsubscribe = useAccountsStore.subscribe(
      (state) => state.isReady,
      (ready) => {
        if (ready) {
          resolve();
          unsubscribe(); // Stop listening after `isReady` becomes true
        }
      }
    );
  });
};
```

Reply:
you guys are right I like this better. It would also resolve sooner (rather than waiting for the 100ms window to elapse)

Reply:
here's a loom with this change: https://www.loom.com/share/d1967df857fb4d1984cbdce195ff7739 (explaining the reason for the wait(1)